### PR TITLE
Add instance ann ~ Void => Pretty (Doc ann)

### PR DESCRIPTION
**Pull request by phadej**
When there are no annotations, Doc is Pretty.

### CODE REVIEW

File: prettyprinter/src/Prettyprinter/Internal.hs [hunk 1]
```diff
@@ -292,6 +293,9 @@ class Pretty a where
 --   , (1, [2, 2, 2, 2, 2, 2, 2, 2])
 --   , (1, [2, 2, 2, 2, 2, 2, 2, 2]) ] ]
 
+instance ann ~ Void => Pretty (Doc ann) where
+  pretty = vacuous
```
Comment (sjakobi) [inline prettyprinter/src/Prettyprinter/Internal.hs:297]: Doesn't `vacuous` perform useless work traversing the `Doc`?! It's implemented as `fmap absurd`.

How about this instead?!

```suggestion
  pretty = unsafeCoerce
```

**Review by sjakobi — COMMENTED** [review]
Thanks! This instance had previously come up in https://github.com/quchen/prettyprinter/issues/227.

The CI failure seems unrelated. I'll try to take care of it soon.

File: prettyprinter/src/Prettyprinter/Internal.hs (hunk 1 above)
Comment (phadej) [reply prettyprinter/src/Prettyprinter/Internal.hs:297]:
```haskell
pretty = unsafeCoerce
```

You forgot a two hundred essay justifying it. 

E.g. I'm not sure what happens when there are bottoms, `annotate (error "surprise") doc :: Doc Void`.

Comment (phadej) [reply prettyprinter/src/Prettyprinter/Internal.hs:297]: To be clear: I won't author a commit with `unsafeCoerce` even `fmap absurd` is almost always can be replaced by it.

**Comment by phadej** [comment]
ping

**Comment by phadej** [comment]
ping

**Pull request merged**